File: nan21.dnet.module.ad/nan21.dnet.module.ad.ui.extjs/src/main/resources/ds/SysDateFormatDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
Ext.define("net.nan21.dnet.module.ad.system.ds.model.SysDateFormatDs" ,{
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"description", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"extjsAltFormats", type:"string"},
		{name:"extjsDateFormat", type:"string"},
		{name:"extjsDateTimeFormat", type:"string"},
		{name:"extjsTimeFormat", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"javaDateFormat", type:"string"},
		{name:"javaDateTimeFormat", type:"string"},
		{name:"javaTimeFormat", type:"string"},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"name", type:"string"},
		{name:"uuid", type:"string"},
		{name:"version", type:"int", useNull:true}
	],
	validations: [
		{field: "name", type: 'presence'}
	]
});
Ext.define("net.nan21.dnet.module.ad.system.ds.model.SysDateFormatDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"description", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"extjsAltFormats", type:"string"},
		{name:"extjsDateFormat", type:"string"},
		{name:"extjsDateTimeFormat", type:"string"},
		{name:"extjsTimeFormat", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"javaDateFormat", type:"string"},
		{name:"javaDateTimeFormat", type:"string"},
		{name:"javaTimeFormat", type:"string"},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"name", type:"string"},
		{name:"uuid", type:"string"}
	]
});
Ext.define("net.nan21.dnet.module.ad.system.ds.param.SysDateFormatDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
});

File: nan21.dnet.module.ad/nan21.dnet.module.ad.presenter/src/main/java/net/nan21/dnet/module/ad/system/ds/model/SysDateFormatDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.ad.system.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeDs;
import net.nan21.dnet.module.ad.system.domain.entity.SysDateFormat;

@Ds(entity = SysDateFormat.class, sort = {@SortField(field = SysDateFormatDs.f_name)})
public class SysDateFormatDs extends AbstractTypeDs<SysDateFormat> {

	public static final String f_extjsDateFormat = "extjsDateFormat";
	public static final String f_extjsTimeFormat = "extjsTimeFormat";
	public static final String f_extjsDateTimeFormat = "extjsDateTimeFormat";
	public static final String f_extjsAltFormats = "extjsAltFormats";
	public static final String f_javaDateFormat = "javaDateFormat";
	public static final String f_javaTimeFormat = "javaTimeFormat";
	public static final String f_javaDateTimeFormat = "javaDateTimeFormat";

	@DsField
	private String extjsDateFormat;

	@DsField
	private String extjsTimeFormat;

	@DsField
	private String extjsDateTimeFormat;

	@DsField
	private String extjsAltFormats;

	@DsField
	private String javaDateFormat;

	@DsField
	private String javaTimeFormat;

	@DsField
	private String javaDateTimeFormat;

	public SysDateFormatDs() {
		super();
	}

	public SysDateFormatDs(SysDateFormat e) {
		super(e);
	}

	public String getExtjsDateFormat() {
		return this.extjsDateFormat;
	}

	public void setExtjsDateFormat(String extjsDateFormat) {
		this.extjsDateFormat = extjsDateFormat;
	}

	public String getExtjsTimeFormat() {
		return this.extjsTimeFormat;
	}

	public void setExtjsTimeFormat(String extjsTimeFormat) {
		this.extjsTimeFormat = extjsTimeFormat;
	}

	public String getExtjsDateTimeFormat() {
		return this.extjsDateTimeFormat;
	}

	public void setExtjsDateTimeFormat(String extjsDateTimeFormat) {
		this.extjsDateTimeFormat = extjsDateTimeFormat;
	}

	public String getExtjsAltFormats() {
		return this.extjsAltFormats;
	}

	public void setExtjsAltFormats(String extjsAltFormats) {
		this.extjsAltFormats = extjsAltFormats;
	}

	public String getJavaDateFormat() {
		return this.javaDateFormat;
	}

	public void setJavaDateFormat(String javaDateFormat) {
		this.javaDateFormat = javaDateFormat;
	}

	public String getJavaTimeFormat() {
		return this.javaTimeFormat;
	}

	public void setJavaTimeFormat(String javaTimeFormat) {
		this.javaTimeFormat = javaTimeFormat;
	}

	public String getJavaDateTimeFormat() {
		return this.javaDateTimeFormat;
	}

	public void setJavaDateTimeFormat(String javaDateTimeFormat) {
		this.javaDateTimeFormat = javaDateTimeFormat;
	}
}
